src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,333 @@
+<?php
+/**
+ * REST API: WP_REST_Global_Styles_Revisions_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.3.0
+ */
+
+/**
+ * Core class used to access global styles revisions via the REST API.
+ *
+ * @since 6.3.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Global_Styles_Revisions_Controller extends WP_REST_Controller {
+	/**
+	 * Parent post type.
+	 *
+	 * @since 6.3.0
+	 * @var string
+	 */
+	private $parent_post_type;
+
+	/**
+	 * The base of the parent controller's route.
+	 *
+	 * @since 6.3.0
+	 * @var string
+	 */
+	private $parent_base;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.3.0
+	 */
+	public function __construct() {
+		$this->parent_post_type = 'wp_global_styles';
+		$this->rest_base        = 'revisions';
+		$this->parent_base      = 'global-styles';
+		$this->namespace        = 'wp/v2';
+	}
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->parent_base . '/(?P<parent>[\d]+)/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'parent' => array(
+						'description' => __( 'The ID for the parent of the revision.' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Returns revisions of the given global styles config custom post type.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param WP_REST_Request $request The request instance.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_items( $request ) {
+		$parent = $this->get_parent( $request['parent'] );
+
+		if ( is_wp_error( $parent ) ) {
+			return $parent;
+		}
+		$response                         = array();
+		$raw_config                       = json_decode( $parent->post_content, true );
+		$is_global_styles_user_theme_json = isset( $raw_config['isGlobalStylesUserThemeJSON'] ) && true === $raw_config['isGlobalStylesUserThemeJSON'];
+
+		if ( $is_global_styles_user_theme_json ) {
+			$user_theme_revisions = wp_get_post_revisions(
+				$parent->ID,
+				array(
+					'posts_per_page' => 100,
+				)
+			);
+
+			if ( ! empty( $user_theme_revisions ) ) {
+				foreach ( $user_theme_revisions as $revision ) {
+					$revision   = $this->prepare_item_for_response( $revision, $request );
+					$response[] = $this->prepare_response_for_collection( $revision );
+				}
+			}
+		}
+
```

Comment:
If no revisions are found, should this not return a WP_Error with a 404 status.

Reply:
I need to refactor this method a bit to support pagination, and will model things on `WP_REST_Revisions_Controller->get_items`.

Looking at `WP_REST_Revisions_Controller->get_items`, it returns an empty array so I'm assuming we should follow that example rather than return an error for consistency.

Reply:
We might, however need a 404 if the `$is_global_styles_user_theme_json` test fails

Reply:
Seems the other get_items methods, do not return an error but an empty array for items.